### DISCUSSION TITLE
Easier .bat file for Offline windows user (with video upscaler)

### DISCUSSION THREAD

**Discussion by unamed666** [xinntao/Real-ESRGAN#767] [Show and tell]
I made .bat file for [**Windows Offline executable version** ](https://github.com/xinntao/Real-ESRGAN?tab=readme-ov-file#portable-executable-files-ncnn)  so people can use most available feature easier without need to understand complex prompt

[Realesrgan.zip](https://github.com/user-attachments/files/17177987/Realesrgan.zip)

## How to use:
- Download [**Realesrgan Windows Offline executable version** ](https://github.com/xinntao/Real-ESRGAN?tab=readme-ov-file#portable-executable-files-ncnn)
- Create a folder named Realesrgan and extract the downloaded file into it
- Download and extract [ my zip file](https://github.com/user-attachments/files/17177987/Realesrgan.zip) into Realesrgan folder, same folder as **_.exe_** file so it will become like this (just ignore gamebanana logo) :
![2024-09-29 13-01-09](https://github.com/user-attachments/assets/1b6a228a-4c72-4eb6-9891-87bdb5c343ae)

- Run the 4.bat or 5.bat
- choose ur **_image_** file
- follow the instruction
- finish

#### Ur upscaled image will be saved in Realesrgan folder

***

**Video upscale** version need [**ffmpeg**](https://github.com/BtbN/FFmpeg-Builds/releases/download/latest/ffmpeg-master-latest-win64-gpl-shared.zip) to work. More tutor below

## How to install video upscale:
- Download [**ffmpeg**](https://github.com/BtbN/FFmpeg-Builds/releases/download/latest/ffmpeg-master-latest-win64-gpl-shared.zip)
- extract .zip file to realesrgan folder, same folder as **_.exe_** file
- _**rename**_ extracted folder to **_ffmpeg_** and it will become like this
![2024-09-29 13-01-32](https://github.com/user-attachments/assets/d734256a-3338-4d27-9fa0-fce1e5885a6f)

***

Now video.bat is ready to use

## How to use video upscale:
- Run 2.bat, choose video u want to upscale, follow the instructions, The **_tmp_frames_** folder should filled with ur video frame
- Run 3.bat, follow the instructions, The **_out_frames_** folder should filled with ur upscaled frame
- Run 4.bat, choose ur video **_again_**, follow the instructions
- finish

## For video please read this 
- Ur upscaled video will be saved in Realesrgan folder
- If u want to use video upscale **_again_**, u need to run 1.bat first to **_clear_** older frame files.
- If you **reupscale** the same video due to delay audio problem or other reason, dont forget to move/rename/delete ur older upscaled video to avoid overwriting problems.
- Sometimes your upscaled video will have different length than the original and it will make the audio delayed based on that lenght. **To fix it**, when running 4.bat fill optional command with :
 -vf "setpts=(X/Y)*PTS"

X = Original length
Y = failed upscaled length

So if original video length is 5 second and failed 5.68 second, type it as :
-vf "setpts=(5/5.68)*PTS"

> Some word in .bat file still using Indonesia language and I just too lazy to edit it 🔢

**Comment by luozi3** [comment]
Thank you, it works perfectly and is very convenient.

**Comment by luozi3** [comment]
Hi, I've noticed that selecting -s 2 when using 5.bat and 6.bat will cause the results to appear as an image blocks, which seems to be caused by the model not supporting X2 and X3.
Or is there something wrong with my usage? I'm a bit confused by this question, in the old version, I remember that I could just use -s 2 to control the magnification

**Reply by unamed666** [reply]
I got this problem too. You see in models folder x2 and x3 only available in video, not normal and anime image, and I just forgot about it XD. But the funny is we can still use anime video for single image, so only normal ver will not have multiple scale
I will update it

EDIT: UPDATED. Anime version scale now work, and no need to set scale for normal version because it only support X4
